### PR TITLE
Test install in GitHub Actions

### PR DESCRIPTION
**Pull request by junghans**
https://github.com/kokkos/kokkos/pull/3686#issuecomment-750381807

/cc @dalg24

### CODE REVIEW

File: .github/workflows/continuous-integration-workflow.yml [hunk 1]
```diff
@@ -58,5 +58,5 @@ jobs:
       - name: Test install
         working-directory: example/tutorial/01_hello_world
         run: |
-          cmake -B builddir
+          cmake -B builddir -DCMAKE_CXX_COMPILER=${{ matrix.cxx }}
```
Comment (junghans) [inline .github/workflows/continuous-integration-workflow.yml:61]: @dalg24 that hints towards an issue, i think we should warn users if they try to using a different compiler than what kokkos was compiled with.

File: .github/workflows/continuous-integration-workflow.yml [hunk 2]
```diff
@@ -55,3 +56,8 @@ jobs:
         run: DESTDIR=${PWD}/install cmake --build builddir --target install && rm -rf ${PWD}/install/usr && rmdir ${PWD}/install
       - name: Install
         run: sudo cmake --build builddir --target install
+      - name: Test install
+        working-directory: example/tutorial/01_hello_world
```
Comment (masterleinad) [inline .github/workflows/continuous-integration-workflow.yml:60]: The Jenkins CI uses `build_cmake_installed` instead:
https://github.com/kokkos/kokkos/blob/af5d246e933065e68ed40d8b1a1e8fccfd7b6ae4/.jenkins#L237-L276

File: .github/workflows/continuous-integration-workflow.yml (hunk 1 above)
Comment (masterleinad) [reply .github/workflows/continuous-integration-workflow.yml:62]: That probably should be discussed. Do you want to open an issue for that?

**Review by masterleinad — APPROVED** [review]
I am fine with the changes here but we could just use the same project for testing as we do in the Jenkins CI.

File: .github/workflows/continuous-integration-workflow.yml (hunk 2 above)
Comment (junghans) [reply .github/workflows/continuous-integration-workflow.yml]: Nice let's use that.

File: .github/workflows/continuous-integration-workflow.yml (hunk 1 above)
Comment (junghans) [reply .github/workflows/continuous-integration-workflow.yml:62]: Meh, I leave that to you ;-)

**Review by dalg24 — APPROVED** [review]

**Pull request merged**